spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultPublisherFactory.java

```diff
@@ -53,11 +55,17 @@
 
 	private final ChannelProvider channelProvider;
 
+	private final CredentialsProvider credentialsProvider;
+
 	public DefaultPublisherFactory(String projectId, ExecutorProvider executorProvider,
-			ChannelProvider channelProvider) {
+			CredentialsProvider credentialsProvider) {
 		this.projectId = projectId;
 		this.executorProvider = executorProvider;
-		this.channelProvider = channelProvider;
+		this.channelProvider = TopicAdminSettings.defaultChannelProviderBuilder()
```

Comment:
I see that the `ChannelProvider` has [multiple implementations](http://googleapis.github.io/gax-java/0.2.0/apidocs/com/google/api/gax/grpc/ChannelProvider.html), but I'm not sure what the use-case would be for anything but the default. Maybe for testing?